modules/git/git.go

```diff
@@ -20,7 +20,36 @@ func GetCurrentBranchName(t testing.TestingT) string {
 
 // GetCurrentBranchNameE retrieves the current branch name or
 // empty string in case of detached state.
+// Uses branch --show-current, which was introduced in git v2.22.
+// Falls back to rev-parse for users of the older version, like Ubuntu 18.04.
 func GetCurrentBranchNameE(t testing.TestingT) (string, error) {
+	cmd := exec.Command("git", "branch", "--show-current")
+	bytes, err := cmd.Output()
+	if err != nil {
+		cmd := exec.Command("git", "rev-parse", "--abbrev-ref", "HEAD")
+		bytes, err := cmd.Output()
+		if err != nil {
+			return "", err
+		}
+		name := strings.TrimSpace(string(bytes))
+		if name == "HEAD" {
+			return "", nil
+		}
+
+		return name, nil
+	}
+
+	name := strings.TrimSpace(string(bytes))
+	if name == "HEAD" {
+		return "", nil
+	}
```

Comment:
NIT: DRY with the logic above by extracting the git subprocess call into its own function.

Reply:
Of course!